src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -656,6 +656,78 @@ static int _write_fix_missing_warning_to_fd(const char *sys, int output_fd, stru
 	}
 }
 
+struct blueprint_entries {
+	const char *pattern;
+	struct oscap_list *list;
+	pcre *re;
+};
+
+static inline int _parse_blueprint_fix(const char *fix_text, struct oscap_list *generic, struct oscap_list *services_enable, struct oscap_list *services_disable, struct oscap_list *kernel_append)
+{
+	const char *err;
+	int errofs;
+	int ret = 0;
+
+	struct blueprint_entries tab[] = {
+		{"\\[customizations\\.services\\]\\s+enabled[=\\s]+\\[([^\\]]+)\\]\\s+", services_enable, NULL},
+		{"\\[customizations\\.services\\]\\s+disabled[=\\s]+\\[([^\\]]+)\\]\\s+", services_disable, NULL},
+		{"\\[customizations\\.kernel\\]\\s+append[=\\s\"]+([^\"]+)[\\s\"]+", kernel_append, NULL},
+		// We do this only to pop the 'distro' entry to the top of the generic list,
+		// effectively placing it to the root of the TOML document.
+		{"\\s+(distro[=\\s\"]+[^\"]+[\\s\"]+)", generic, NULL},
+		{NULL, NULL, NULL}
+	};
+
+	for (int i = 0; tab[i].pattern != NULL; i++) {
+		tab[i].re = pcre_compile(tab[i].pattern, PCRE_UTF8, &err, &errofs, NULL);
+		if (tab[i].re == NULL) {
+			dE("Unable to compile /%s/ regex pattern, pcre_compile() returned error (offset: %d): '%s'.\n", tab[i].pattern, errofs, err);
+			ret = 1;
+			goto exit;
+		}
+	}
+
+	const size_t fix_text_len = strlen(fix_text);
+	size_t start_offset = 0;
+	int ovector[10] = {0};
```

Comment:
The vector size should be a multiple of 3, but the man page says that if it isn't it will get rounded down, so I'm not sure if it is a problem.

Reply:
Nah, not a problem. But I'll align it to the current number of capture groups.